### PR TITLE
NIO Cleanup Refactoring

### PR DESCRIPTION
**Pull request by DanVanAtta**
## Overview
- Remove unused constructor
- Replace methods with equivalent lombok annotation
- Move logging of socket data to clear up high level flow.
- Rename method 'closed' -> 'close'
- Simplify: Remove threadSuffix name parameter
- Remove 'broadcast' method called in test-only

### CODE REVIEW

**Comment by RoiEXLab** [comment]
@DanVanAtta The build is failing due to compile errors

File: game-core/src/main/java/games/strategy/net/IMessenger.java [hunk 1]
```diff
@@ -16,11 +16,6 @@
    */
   void send(Serializable msg, INode to);
 
-  /**
-   * Send a message to all nodes.
-   */
-  void broadcast(Serializable msg);
-
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/IMessenger.java]: I'm surprised this method isn't used. How does chat broadcast the messages if not via this method?

File: game-core/src/main/java/games/strategy/net/nio/SocketReadData.java [hunk 2]
```diff
@@ -16,19 +16,19 @@
...
+  static final int MAGIC = 0x9b000000;
 
   private int targetSize = -1;
   // we read into here the first four bytes to find out size
   private ByteBuffer sizeBuffer;
   // we read into here after knowing out size
   private ByteBuffer contentBuffer;
+  @Getter
   private final SocketChannel channel;
-  private final int number = counter.incrementAndGet();
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/SocketReadData.java]: Good spot 👍

File: game-core/src/main/java/games/strategy/net/IMessenger.java (hunk 1 above)
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/net/IMessenger.java]: Nullable state is used:
```
@Getter
@AllArgsConstructor
public class MessageHeader {
  // to can be null if we are a broadcast
  @Nullable
  private final INode to;
  // from can be null if the sending node doesnt know its own address
  @Nullable
  private final INode from;
  private final Serializable message;
}
```

File: game-core/src/main/java/games/strategy/net/nio/NioReader.java [hunk 3]
```diff
@@ -134,6 +124,20 @@ private void loop() {
...
+    SocketAddress sa = null;
+    if (s != null) {
+      sa = s.getRemoteSocketAddress();
+    }
+    String remote = "null";
+    if (sa != null) {
+      remote = sa.toString();
+    }
+    log.fine(" done reading from:" + remote + " size:" + packet.size() + " readCalls;"
+        + packet.getReadCalls() + " total:" + totalBytes);
```
[the remark this answers is not in the document]
Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/net/nio/NioReader.java]: You're right, though it's even better to just remove this code. There is further simplifications, local state, that can be removed, *and* there is a duplicated version in the writer: b899cf8846bfabfa5608b7ae530adb3cdda72793

On another hand, this code is really debug level. It's bad practice to leave this level of debug in as it is cluttering.

File: game-core/src/main/java/games/strategy/net/IMessenger.java (hunk 1 above)
Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/net/IMessenger.java]: Ok wow.
Legacy code strikes again. 😥 
I'm not even surprised anymore.

File: game-core/src/main/java/games/strategy/net/nio/NioReader.java [hunk 4]
```diff
@@ -100,20 +97,6 @@ private void loop() {
...
-                  if (s != null) {
-                    sa = s.getRemoteSocketAddress();
-                  }
-                  String remote = "null";
-                  if (sa != null) {
-                    remote = sa.toString();
-                  }
-                  log.fine(" done reading from:" + remote + " size:" + packet.size() + " readCalls;"
-                      + packet.getReadCalls() + " total:" + totalBytes);
-                }
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/net/nio/NioReader.java]: That's even better 👍

**Pull request merged**